### ISSUE TITLE
2.7-alpine bundler install location not in gem environment

### ISSUE THREAD

**Issue by rrazor** [docker-library/ruby#305]
**Overview**
Gems installed via Bundler in the `2.7-alpine` series of images will not install to a directory that is known to the Gems environment.

**Steps**

1. Obtain a shell:

        $ TAG=2.7-alpine
        $ docker run -it ruby:$TAG sh
        / #

2. Create a basic `Gemfile`:

        / # cat > Gemfile <<EOF
        source 'https://rubygems.org'
        gem 'rspec'
        EOF

3. Install with Bundler:

        / # bundle install
        [...]
        Bundled gems are installed into `./usr/local/bundle`

4. Test with `irb`:

        / # irb
        irb(main):001:0> require 'rspec'

**Expected results**

    / # irb
    irb(main):001:0> require 'rspec'
    => true

**Actual results**
For `TAG=2.7-alpine`, the following error occurs:

     / # irb
     irb(main):001:0> require 'rspec'
     Traceback (most recent call last):
             6: from /usr/local/bin/irb:23:in `<main>'
             5: from /usr/local/bin/irb:23:in `load'
             4: from /usr/local/lib/ruby/gems/2.7.0/gems/irb-1.2.1/exe/irb:11:in `<top (required)>'
             3: from (irb):1
             2: from /usr/local/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
             1: from /usr/local/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
     LoadError (cannot load such file -- rspec)

**Technical**
It seems that the Bundler gem install location in the 2.7 version is not in the Gem environment:

```
/ # gem environment
RubyGems Environment:
[...]
  - GEM PATHS:
     - /usr/local/bundle
     - /root/.gem/ruby/2.7.0
     - /usr/local/lib/ruby/gems/2.7.0
[...]
```

Note the absence of `/usr/local/bundle/ruby/2.7.0` in the `GEM PATHS` above.

**Comment by rrazor** [comment]
Looks like Bundler behavior changed:
https://github.com/bundler/bundler/issues/7494

**Comment by wglambert** [comment]
Looks like we're waiting for the decision from @deivid-rodriguez on either a change in the Docker image or an upstream Bundler change

https://github.com/docker-library/official-images/pull/7208#issuecomment-569312702
>See https://github.com/docker-library/official-images/pull/7166 -- this was an intentional change by @deivid-rodriguez from Bundler upstream to try and help avoid https://github.com/bundler/bundler/issues/7494 from cropping back up when we get either the images or Bundler fixed accordingly. :sweat_smile:

https://github.com/docker-library/official-images/pull/7166#issuecomment-568143574
>The current failures are expected, and will be relevant either way. They will be fixed either by modifying the images, or in the next patch level version of bundler if we end up fixing this in `bundler`. I'll give it some more thought in the next days, since I'm still on the fence, will post my conclusions here and follow up with patches and more tests.

**Comment by wglambert** [comment]
I believe this is the upstream PR being considered https://github.com/bundler/bundler/pull/7501 as one of the potential options

**Comment by deivid-rodriguez** [comment]
This should've been fixed by #306.

**Comment by rrazor** [comment]
@deivid-rodriguez I can confirm that this is fixed for the above repro steps in `master` as of e30082fa0ddb6ee6c03ce2fac005725a0612c2cd. Thank you!
